### DISCUSSION TITLE
InvalidOperationException: The property 'Company.Id' could not be mapped because it is of type 'CompanyId', which is not a supported primitive type or a valid entity type

### DISCUSSION THREAD

**Discussion by MrYossu** [SteveDunn/Vogen#728] [General]
Forgive me if this is a dumb question, but I'm just trying Vogen, and am having a problem.

I created an ASP.NET Core Blazor web app (not sure if that's actually relevant, but who knows), and added an EF Core `DbContext` with a couple of entity types. I got a basic CRUD app running without any value types, just to get going.

I then added Vogen, and added the following value object...

```c#
[ValueObject]
public partial struct CompanyId;
```

I modified my `Company` model to use this instead of `int` for the `Id`...

```c#
public class Company {
  public CompanyId Id { get; set; }
  public string Name { get; set; } = "";
  public string Address { get; set; } = "";
  public string Telephone { get; set; } = "";
}
```

I changed all relevant occurrences of `int` in the code to use `CompanyId`, and added the `VogenExtensions` class shown in the [Tips for working with EF Core wiki page](https://stevedunn.github.io/Vogen/efcore-tips.html#adding-all-value-objects-to-the-modelconfigurationbuilder).

I also added the override for `ConfigureConventions` as shown on the same page.

I then ran my app and navigated to the page that loads companies from the database, but got an internal server error...

>InvalidOperationException: The property 'Company.Id' could not be mapped because it is of type 'CompanyId', which is not a supported primitive type or a valid entity type

Anyone able to explain what I've done wrong? If there's any more code I can show, please let me know. Thanks

**Comment by MrYossu** [comment]
Duh, realised that the problem was due to me not setting up AutoMapper to map the VO.

Closing this before anyone sees it!

**Reply by SteveDunn** [reply]
Glad you got it sorted. Thanks for trying Vogen!

**Reply by MrYossu** [reply]
@SteveDunn Thank you for making it! I've been wanting to try VOs ever since I read Scott Wlaschin's book on DDD. Your library took the grunt work away from me 😎
